design/benchmark_tests.md

```diff
@@ -0,0 +1,33 @@
+# Benchmarks Tests
+
+## Requirements
+1. Measure the latency percentiles (50th, 90th, 95th, 99th 99.9) for a batch of repeated basic operations (e.g. 10K put requests),
+   at a predefined parallelism level.
+2. Show trends of benchmarks tests execution times across different versions.
+3. Portability: The tests don't assume anything on the environment where the lakeFS & DB are running.
+   lakeFS address is the only **required** argument for running the tests.
+
+## Non-Requirements
+1. An active gatekeeper - single slow running cycle shouldn't block the build.
+2. Expect 100% reliability of all actions - when performing thousands of operations,
+   some (exact percent TBD) will be allowed to fail. When measuring an external app,
+   it's the app responsibility to retry on transient errors.
+3. Avoid repeating unit/integration/system tests scenarios.
+  
+## Solution
+### Test Execution
+For each parallelism level, the flow will create a repository and will PUT 10k files in it.
+Then, it will try to read those files. Finally, it will delete them.
```

Comment:
I agree the focus should be on critical path operations (i.e. reads and writes).
I am missing however:

1. Common prefix listing (which is also a very common critical path operation)
2. Anther scenario where we preform all said operations on an arbitrarily nested branch. Can be as simple as committing after the initial execution, creating a branch derived from master and repeating the experiment on that branch.

Reply:
Done.